vortex-array/src/accessor.rs

```diff
@@ -1,7 +1,13 @@
 use vortex_error::VortexResult;
 
+/// Trait for arrays that support iterative access to their elements.
 pub trait ArrayAccessor<Item: ?Sized> {
```

Comment:
Meta note: This interface should be changed. We should have accessors and iterator independetly of each other so you can efficiently access element n without going through scalar_at